File: Prosary/Models/EternalRestPlacement.swift
//
//  EternalRestPlacement.swift
//  Prosary
//

import Foundation

/// Where (if at all) "Eternal rest grant unto them, O Lord" is prayed for the faithful departed.
enum EternalRestPlacement: String, Codable, CaseIterable, Identifiable {
  case none
  /// Pray it after the Glory Be of every decade.
  case afterEachDecade
  /// Pray it once, near the end, before the closing prayers.
  case atEndOnly

  var id: String { rawValue }

  var displayName: String {
    switch self {
    case .none:           return String(localized: "Don't Include")
    case .afterEachDecade: return String(localized: "After Each Decade")
    case .atEndOnly:      return String(localized: "Once, Near the End")
    }
  }
}

File: Prosary/Models/EternalRestPlacement.cs
namespace Prosary.Models;

public enum EternalRestPlacement
{
    None,

    /// <summary>Pray "Eternal rest grant unto them, O Lord" after the Glory Be of every decade.</summary>
    AfterEachDecade,

    /// <summary>Pray it once, near the end, before the closing prayers.</summary>
    AtEndOnly
}

public static class EternalRestPlacementExtensions
{
    public static string DisplayName(this EternalRestPlacement placement) => placement switch
    {
        EternalRestPlacement.None => "Don't Include",
        EternalRestPlacement.AfterEachDecade => "After Each Decade",
        EternalRestPlacement.AtEndOnly => "Once, Near the End",
        _ => throw new ArgumentOutOfRangeException(nameof(placement))
    };
}
